contrib/babelfishpg_tsql/sql/sys_views.sql

```diff
@@ -3219,19 +3297,111 @@ FROM sys.events e
+    indexrelid,
+    case
+      when indisclustered then 1
+      else 1+row_number() over(partition by indrelid order by indexrelid)
+    end as index_id
+  from pg_index
+),
+tt_internal as MATERIALIZED
+(
+  select * from sys.table_types_internal
+)
+-- entries for non-partitioned tables
+SELECT
+  CAST(t.oid as sys.BIGINT) as partition_id,
+  CAST(t.oid as int) as object_id,
+  CAST(0 as int) as index_id,
+  CAST(1 as int) as partition_number,
+  CAST(0 as sys.bigint) AS hobt_id,
+  CAST(case when t.reltuples = -1 then 0 else t.reltuples end as sys.bigint) AS rows,
+  CAST(0 as smallint) as filestream_filegroup_id,
+  CAST(0 as sys.tinyint) as data_compression,
+  CAST('NONE' as sys.nvarchar(60)) as data_compression_desc,
+  CAST(0 as sys.bit) as xml_compression,
+  CAST('OFF' as sys.varchar(3)) as xml_compression_desc
+FROM pg_class t
+INNER JOIN sys.schemas sch on sch.schema_id = t.relnamespace
+LEFT JOIN tt_internal tt on t.oid = tt.typrelid
+WHERE tt.typrelid is null
+AND t.relkind = 'r'
+AND t.relispartition = false
+AND has_schema_privilege(t.relnamespace, 'USAGE')
+AND has_table_privilege(t.oid, 'SELECT,INSERT,UPDATE,DELETE,TRUNCATE,TRIGGER')
+
+UNION ALL
+-- entries for non-partitioned indexes
+SELECT
+  CAST(idx.indexrelid as sys.BIGINT) as partition_id,
+  CAST(idx.indrelid as int) as object_id,
+  CAST(imap.index_id as int) as index_id,
+  CAST(1 as int) as partition_number,
+  CAST(0 as sys.bigint) AS hobt_id,
+  CAST(case when t.reltuples = -1 then 0 else t.reltuples end as sys.bigint) AS rows,
+  CAST(0 as smallint) as filestream_filegroup_id,
+  CAST(0 as sys.tinyint) as data_compression,
+  CAST('NONE' as sys.nvarchar(60)) as data_compression_desc,
+  CAST(0 as sys.bit) as xml_compression,
+  CAST('OFF' as sys.varchar(3)) as xml_compression_desc
+FROM pg_index idx
+INNER JOIN index_id_map imap on imap.indexrelid = idx.indexrelid
+INNER JOIN pg_class t on t.oid = idx.indrelid and t.relkind = 'r' and t.relispartition = false
+INNER JOIN sys.schemas sch on sch.schema_id = t.relnamespace
+where idx.indislive
+and has_schema_privilege(t.relnamespace, 'USAGE')
+
+UNION ALL
+-- entries for partitions of partitioned tables
+SELECT
+  CAST(pgi.inhrelid as sys.BIGINT) as partition_id,
+  CAST(pgi.inhparent as int) as object_id,
+  CAST(0 as int) as index_id,
+  CAST(row_number() over(partition by pgi.inhparent order by ctbl.relname) as int) as partition_number,
+  CAST(0 as sys.bigint) AS hobt_id,
+  CAST(case when ctbl.reltuples = -1 then 0 else ctbl.reltuples end as sys.bigint) AS rows,
+  CAST(0 as smallint) as filestream_filegroup_id,
+  CAST(0 as sys.tinyint) as data_compression,
+  CAST('NONE' as sys.nvarchar(60)) as data_compression_desc,
+  CAST(0 as sys.bit) as xml_compression,
+  CAST('OFF' as sys.varchar(3)) as xml_compression_desc
+FROM pg_inherits pgi
+INNER JOIN pg_class ctbl on ctbl.oid = pgi.inhrelid
+INNER JOIN pg_class ptbl on ptbl.oid = pgi.inhparent
+INNER JOIN sys.schemas sch on sch.schema_id = ptbl.relnamespace
```

Comment:
Do we need sys.schemas here? Why cant we be more efficient/cleaner by using babelfish_namespace_ext and db_id

Reply:
Same for all the usages of sys.schemas elsewhere, please re-check

Reply:
addressed everywhere.